modules/jolt_physics/register_types.cpp

```diff
@@ -0,0 +1,79 @@
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "register_types.h"
+
+#include "jolt_globals.h"
+#include "jolt_physics_server_3d.h"
+#include "jolt_project_settings.h"
+
+#include "servers/physics_server_3d_wrap_mt.h"
+
+PhysicsServer3D *create_jolt_physics_server() {
+#ifdef THREADS_ENABLED
+	bool run_on_separate_thread = GLOBAL_GET("physics/3d/run_on_separate_thread");
+#else
+	bool run_on_separate_thread = false;
+#endif
+
+	JoltPhysicsServer3D *physics_server = memnew(JoltPhysicsServer3D(run_on_separate_thread));
+
+	return memnew(PhysicsServer3DWrapMT(physics_server, run_on_separate_thread));
+}
+
+void initialize_jolt_physics_module(ModuleInitializationLevel p_level) {
+	switch (p_level) {
+		case MODULE_INITIALIZATION_LEVEL_CORE: {
+		} break;
+		case MODULE_INITIALIZATION_LEVEL_SERVERS: {
+			jolt_initialize();
+			PhysicsServer3DManager::get_singleton()->register_server("JoltPhysics3D", callable_mp_static(&create_jolt_physics_server));
```

Comment:
Sure, I'd fine with dropping it. I mostly put it there to conform with Godot Physics, but I never quite understood why the suffix was there for Godot Physics to begin with, since the 2D and 3D servers are registered using different managers anyway, and never listed together.

Do we drop the suffix from the `physics/jolt_physics_3d/...` settings paths as well then?

And what about the class names? Most of them are suffixed with `3D` as well, which wasn't the case for the Bullet physics server. But then again, none of the 3D stuff had the `3D` suffix in Godot 3.x, so maybe that convention still applies.

Reply:
> And what about the class names? Most of them are suffixed with `3D` as well, which wasn't the case for the Bullet physics server. But then again, none of the 3D stuff had the `3D` suffix in Godot 3.x, so maybe that convention still applies.

Reduz wanted to make the distinction between 2D and 3D class as unambiguous as possible in the transition to Godot 4. Classes that are shared between both have the suffix. You can argue the same idea applies to the physics servers.

Reply:
Technically Jolt could also be used to implement a 2D physics server. I'm not sure how future proof we want to be, but keeping the 3D postfix leaves that possibility open?

Reply:
I went ahead and dropped the `3D` suffix for the listed server name, since there shouldn't be a conflict there even if we did implement some 2D server in the future.

I've left the rest as-is.